packages/fabrix/src/renderers/fields.tsx

```diff
@@ -85,42 +93,23 @@ export const ViewRenderer = ({
     : renderFields();
 };
 
-/**
- * Get the type name of the given field by looking at the __typename field.
- */
-const getTypeName = (
-  context: FabrixContextType,
-  rootValue: Value | undefined,
-  name: string,
-) => {
-  if (Array.isArray(rootValue)) {
-    return resolveFieldTypesFromTypename(context, rootValue[0][name]);
-  } else if (typeof rootValue?.[name] === "object") {
-    return resolveFieldTypesFromTypename(context, rootValue?.[name]);
-  } else {
-    return {};
-  }
-};
-
 /**
  * Get the sub fields of the given field.
  *
  * This also sorts the fields by the index value.
  */
 export const getSubFields = (
-  context: FabrixContextType,
-  rootValue: Value | undefined,
+  typenameExtractor: TypenameExtractor,
   fields: ViewFields,
   name: string,
 ) =>
   // filters fields by parent key and maps the filtered values to the array of SubField
   fields
-    .filter((f) => f.field.getParent()?.asKey() === name)
+    .filter((f) => f.field.getParent()?.asKey().startsWith(name))
```

Comment:
Here is the key to support nested field selection. 

Previously here filters out with the exact match of parent name, but using `startsWith` does not limit the children to filter.